esp/bindings/http/platform/httpservice.cpp

```diff
@@ -1429,9 +1435,29 @@ EspAuthState CEspHttpServer::handleUserNameOnlyMode(EspAuthRequest& authReq)
     if (!userName.isEmpty())
         return authSucceeded;
 
+    bool invalidUserName = false;
     const char* userNameIn = (authReq.requestParams) ? authReq.requestParams->queryProp("username") : NULL;
-    if (isEmptyString(userNameIn))
+    if (!isEmptyString(userNameIn))
+        invalidUserName = !isValidUserName(userNameIn);
+    if (isEmptyString(userNameIn) || invalidUserName)
```

Comment:
Maybe out of scope, but thinking of validity as I was made me ask how a name containing only whitespace should be viewed. Can it happen? Should the name be trimmed before testing for empty? Should such a name be consider malformed? Do we care, as long as it doesn't contain any characters that we could choke on?

Reply:
It can happen. The existing code handles it when adding an empty cookie and a user will be asked to login again. It now works when the name is trimmed and tested inside the isMalformedUserName().